data/src/test/java/com/linkedin/data/codec/symbol/TestDefaultSymbolTableProvider.java

```diff
@@ -0,0 +1,105 @@
+
+  @Test
+  public void testRemoteSymbolTableSuccess() throws Exception
+  {
+    ByteString serializedTable = SymbolTableSerializer.toByteString(DefaultSymbolTableProvider.CODEC, _symbolTable);
+
+    HttpURLConnection connection = mock(HttpURLConnection.class);
+    DefaultSymbolTableProvider provider = spy(new DefaultSymbolTableProvider());
+    doReturn(connection).when(provider).openConnection(eq("https://someservice:100/symbolTable/tableName"));
+    when(connection.getResponseCode()).thenReturn(200);
+    when(connection.getInputStream()).thenReturn(serializedTable.asInputStream());
+
+    SymbolTable remoteTable = provider.getSymbolTable(_symbolTableName);
+    verify(connection).setRequestProperty(eq("Accept"), eq(ProtobufDataCodec.DEFAULT_HEADER));
+
+    // Verify table is deserialized correctly.
+    Assert.assertEquals(_symbolTable, remoteTable);
+
+    // Mock out the network to throw exceptions on any interactions.
+    doThrow(new RuntimeException()).when(provider).openConnection(anyString());
+
+    // Verify that table is in cache by retrieving it again.
+    Assert.assertEquals(provider.getSymbolTable("tableName"), remoteTable);
+  }
+
+  @Test(expectedExceptions = IllegalStateException.class)
+  public void testRemoteSymbolTableMalformedUrl()
+  {
+    String symbolTableName = "https\\someservice:100|tableName";
+    new DefaultSymbolTableProvider().getSymbolTable(symbolTableName);
+  }
+
+  @Test(expectedExceptions = IllegalStateException.class)
+  public void testRemoteSymbolTableErrorResponseCode() throws Exception
+  {
+    HttpURLConnection connection = mock(HttpURLConnection.class);
+    DefaultSymbolTableProvider provider = spy(new DefaultSymbolTableProvider());
+    doReturn(connection).when(provider).openConnection(eq("https://someservice:100/symbolTable/tableName"));
+    when(connection.getResponseCode()).thenReturn(500);
+    provider.getSymbolTable(_symbolTableName);
+  }
+
+  @Test(expectedExceptions = IllegalStateException.class)
+  public void testRemoteSymbolTableParsingFailure() throws Exception
+  {
+    ByteString serializedTable = ByteString.unsafeWrap("random".getBytes());
+
+    HttpURLConnection connection = mock(HttpURLConnection.class);
+    DefaultSymbolTableProvider provider = spy(new DefaultSymbolTableProvider());
+    doReturn(connection).when(provider).openConnection(eq("https://someservice:100/symbolTable/tableName"));
+    when(connection.getResponseCode()).thenReturn(200);
+    when(connection.getInputStream()).thenReturn(serializedTable.asInputStream());
+
+    provider.getSymbolTable(_symbolTableName);
+    verify(connection).setRequestProperty(eq("Accept"), eq(ProtobufDataCodec.DEFAULT_HEADER));
+  }
+
+  @Test
+  public void testLocalSymbolTableSuccess()
+  {
+    DefaultSymbolTableProvider provider = new DefaultSymbolTableProvider();
+    // Inject the table into the local cache.
+    provider._cache.put("tableName", _symbolTable);
```

Comment:
minor: Do you think there will be usecase to inject local symbol table? If so consider adding an API to do that.

Reply:
Good idea. Added.